src/Symfony/Component/Console/Helper/TreeNode.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+/**
+ * @implements \IteratorAggregate<TreeNode>
+ *
+ * @author Simon André <smn.andre@gmail.com>
+ */
+final class TreeNode implements \Countable, \IteratorAggregate
+{
+    /**
+     * @var array<TreeNode|callable(): \Generator>
+     */
+    private array $children = [];
+
+    public function __construct(
+        private readonly string $value = '',
+        ?self $parent = null,
```

Comment:
do we need this? can't this be moved as an outside concern?

Reply:
😶‍🌫️ ux habit  

(removed)

Reply:
Done, thx